keps/sig-security/2763-ambient-capabilities/README.md

```diff
@@ -0,0 +1,1066 @@
+
+- We also propose that container runtimes like `containerd` and `CRI-O` be updated to account for these changes to the CRI apis and add the requested capabilities to the ambient set when they create the containers. Details on how and where these changes will be made are explained in the [Changes to containerd/containerd (<a href="https://github.com/containerd/containerd">https://github.com/containerd/containerd</a>)](#changes-to-containerdcontainerd-httpsgithubcomcontainerdcontainerd) section below.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a security conscious user I should be able to run my container as non-root and add the minimum set of capabilities required for it to function even when I do not control how the image is built.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Changes to kubernetes API (https://pkg.go.dev/k8s.io/api/core/v1)
+
+<<[UNRESOLVED should we reuse add instead of adding new field]>>
+
+Reusing the existing add filed in the Capabilities field in the securityContext might cause confusion and we propose adding a new field called `Ambient` to the [Capabilities](https://pkg.go.dev/k8s.io/api/core/v1#Capabilities) struct. 
+
+```diff
+type Capabilities struct {
+  // Added capabilities
+  // +optional
+  Add []Capability `json:"add,omitempty" protobuf:"bytes,1,rep,name=add,casttype=Capability"`
```

Comment:
Does `Add` serve a meaningful purpose in the API once `Ambient` exists?  Could it be deprecated once `Ambient` is GA?

Reply:
good point, I think we could once Ambient is in GA.

Reply:
Maybe `Add` make sense when you actually want to retain old behavior and rely on capabilities added during the image building phase?

Reply:
I am thinking we can dedicate `Add` to root user case, as it currently does.